### PR TITLE
Add bounds check when drawing line high/low

### PR DESCRIPTION
**Pull request by DarrienG**
Fix for underflow bug when using line generation: https://github.com/fdehau/tui-rs/issues/280

### CODE REVIEW

File: src/widgets/canvas/line.rs [hunk 1]
```diff
@@ -64,7 +64,7 @@ fn draw_line_low(painter: &mut Painter, x1: usize, y1: usize, x2: usize, y2: usi
     for x in x1..=x2 {
         painter.paint(x, y, color);
         if d > 0 {
-            y = if y1 > y2 { y - 1 } else { y + 1 };
+            y = if y1 > y2 && y1 > 0 { y - 1 } else { y + 1 };
```
Comment (extrawurst) [inline src/widgets/canvas/line.rs:67]: what about the *theoretical* overflow case of `y+1` ? would it be clearer communication-wise to use `saturated_add`/`saturated_sub` here?

Comment (ClementTsang) [reply src/widgets/canvas/line.rs:67]: I personally also think just using `saturated_sub/saturated_add` will do the trick.

Comment (DarrienG) [reply src/widgets/canvas/line.rs:67]: TIL these exist. I'll update after I finish up after work. Thanks for the suggestions!

Comment (DarrienG) [reply src/widgets/canvas/line.rs]: Saturated!

**Comment by fdehau** [comment]
Thanks for reporting the issue. The fix makes sense but could you also add a test to cover this edge case ?

**Comment by DarrienG** [comment]
Totally! I'm honestly not certain how the code path is hit, but I added a test that recreates the conditions I got the crash. If there's a better way to trigger the error, I can revise :)
